pkg/controller/scheduler.go

```diff
@@ -318,6 +318,19 @@ func (c *Controller) advanceCanary(name string, namespace string) {
 
 	// check if canary revision changed during analysis
 	if restart := c.hasCanaryRevisionChanged(cd, canaryController); restart {
+		// update canary status to ensure checksum is up to date in all webhooks
+		if err := canaryController.SyncStatus(cd, flaggerv1.CanaryStatus{Phase: flaggerv1.CanaryPhaseProgressing}); err != nil {
```

Comment:
Instead of calling `SyncStatus` twice for a restart, I think we could move the `recordEventInfof` at the end

Reply:
@stefanprodan sure that makes sense, done.